packages/engine/src/mocap/MotionCaptureActions.ts

```diff
@@ -23,13 +23,41 @@ All portions of the code written by the Ethereal Engine team are Copyright © 20
 Ethereal Engine. All Rights Reserved.
 */
 
-import { defineAction } from '@etherealengine/hyperflux'
+import { EntityUUID } from '@etherealengine/common/src/interfaces/EntityUUID'
+import { defineAction, defineState } from '@etherealengine/hyperflux'
 import matches from 'ts-matches'
+import { ikTargets } from '../avatar/animation/Util'
+import { AvatarIKTargetComponent } from '../avatar/components/AvatarIKComponents'
+import { setComponent } from '../ecs/functions/ComponentFunctions'
+import { NetworkState } from '../networking/NetworkState'
+import { NetworkTopics } from '../networking/classes/Network'
+import { UUIDComponent } from '../scene/components/UUIDComponent'
 
 export class MotionCaptureAction {
   static trackingScopeChanged = defineAction({
     type: 'ee.mocap.trackLowerBody' as const,
     trackingLowerBody: matches.boolean,
-    $cache: { removePrevious: true }
+    $topic: NetworkTopics.world
   })
 }
+
+export const MotionCaptureState = defineState({
+  name: 'MotionCaptureState',
+  initial: {},
+  receptors: [
+    [
+      //make this mutate data of the ik targets
+      MotionCaptureAction.trackingScopeChanged,
+      (state, action: typeof MotionCaptureAction.trackingScopeChanged.matches._TYPE) => {
```

Comment:
even sourcing should set a state, rather than having side effects in the receptor, and have reactors react to that state. we should also keep { $cache: removePrevious }